numpy/core/src/umath/loops_modulo.dispatch.c.src

```diff
@@ -329,7 +343,11 @@ vsx4_simd_@func@_contig_@sfx@(char **args, npy_intp len)
 #if @id@ == 2 /* divmod */
         npyv_@sfx@ to_sub = npyv_select_@sfx@(or, vzero, vneg_one);
                       quo = npyv_add_@sfx@(quo, to_sub);
-        npyv_store_@sfx@(dst1, npyv_and_@sfx@(cvtozero, quo));
+                      // Divide by zero
+                      quo = npyv_select_@sfx@(bzero, vzero, quo);
+                      // Overflow
+                      quo = npyv_select_@sfx@(overflow, vmin, quo);
+        npyv_store_@sfx@(dst1, quo);
         npyv_store_@sfx@(dst2, npyv_and_@sfx@(cvtozero, rem));
```

Comment:
Hi @ganesh-k13 , see below one change needed for VSX4/Power10 as the variable/vector `cvtozero` was deleted in one of the commits.

One more thing: your tests are passing when I execute them on Power/VSX.

Thank you for applying the changes for Power/VSX as well :-) :+1: .

P.S.: I realized that the other `cvtozero` that is used in the other function that implements the unsigned integer types could also be removed to reduce 2 SIMD instructions into just one  `(select+and --> select)`.

```suggestion
                      // Divide by zero
                      quo = npyv_select_@sfx@(bzero, vzero, quo);
                      rem = npyv_select_@sfx@(bzero, vzero, rem);
                      // Overflow
                      quo = npyv_select_@sfx@(overflow, vmin, quo);
                      rem = npyv_select_@sfx@(overflow, vzero, rem);
        npyv_store_@sfx@(dst1, quo);
        npyv_store_@sfx@(dst2, rem);
```

Reply:
Thanks a lot @rafaelcfsousa! I have applied the changes to both signed and unsinged, can you take a look once please?

Reply:
Thank you for applying the changes!! :+1:

There are two other small changes for Power/VSX as shown below.

---

1) We also need the addition of the line 

```
rem = npyv_select_@sfx@(bzero, vzero, rem);
```

after the line below as shown in the `suggested change` box in my previous message (above).
https://github.com/numpy/numpy/blob/4f428f0571fb1d4b1f9fcb62ca9013fcd850f918/numpy/core/src/umath/loops_modulo.dispatch.c.src#L347

It is required as we compute the rem of the operation `divmod` using the equation below:

```
rem = a - (b * quo)
```
which means that we have to zero `rem` for `b` equal to zero, otherwise `rem` results in `a`.

---

2. About the variable `cvtozero`, the changes worked perfectly for both signed and unsigned types. But I forgot to say that one other variable (`vneg_one`) would become unused after removing `cvtozero`, see below:

```
numpy/core/src/umath/loops_modulo.dispatch.c.src:184:19: warning: unused variable ‘vneg_one’ [-Wunused-variable]
  184 |     const npyv_@sfx@ vneg_one = npyv_setall_@sfx@(-1);
```

You can remove the variable `vneg_one` defined at numpy/core/src/umath/loops_modulo.dispatch.c.src:184 (see below).
https://github.com/numpy/numpy/blob/4f428f0571fb1d4b1f9fcb62ca9013fcd850f918/numpy/core/src/umath/loops_modulo.dispatch.c.src#L184

---

That's all, thanks again for applying the changes for VSX4/Power10. :-) :+1:

Reply:
Ah ok, so we perform the combined action of `cvtozero` using the two selects. Thanks a lot Rafael! Really shooting in the dark here without the hardware.